packages/ui-components/src/components/DateTimePicker/DateTimePicker.component.tsx

```diff
@@ -0,0 +1,651 @@
+          <Label
+            text={label}
+            htmlFor={theId}
+            className={`${labelStyles}`}
+            disabled={disabled}
+            required={required}
+            floating
+            minimized={
+              placeholder || isOpen || theDate.selectedDate?.length || theDate.selectedDateStr?.length ? true : false
+            }
+          />
+        ) : (
+          ""
+        )}
+
+        <div className={`juno-datetimepicker-icon-container ${iconContainerStyles}`}>
+          {theDate.selectedDate?.length || theDate.selectedDateStr?.length ? (
+            <Icon icon="close" onClick={handleClearIconClick} disabled={disabled} title="Clear" />
+          ) : (
+            ""
+          )}
+          {isInvalid ? <Icon icon="dangerous" color="jn-text-theme-error" /> : ""}
+          {isValid ? <Icon icon="checkCircle" color="jn-text-theme-success" /> : ""}
+        </div>
+      </div>
+      <div ref={calendarTargetRef}></div>
+
+      {errortext && hasLength(errortext) ? <FormHint text={errortext} variant="error" className="jn-mt-0" /> : ""}
+      {successtext && hasLength(successtext) ? (
+        <FormHint text={successtext} variant="success" className="jn-mt-0" />
+      ) : (
+        ""
+      )}
+      {helptext && hasLength(helptext) ? <FormHint text={helptext} className="jn-mt-0" /> : ""}
+    </div>
+  )
+}
+
+export interface DateTimePickerProps
+  extends Omit<React.HTMLAttributes<HTMLInputElement>, "defaultValue" | "onFocus" | "onBlur" | "onChange"> {
+  /** Whether the DateTimePicker input element allows direct user keyboard input. Default is `false`. */
+  allowInput?: boolean
+  /** Allows the preloading of an invalid date (e.g. a date that hass been disable by passing `disable`). When disabled, the field will be cleared if the provided date is invalid */
+  allowInvalidPreload?: boolean
+  /** How the `aria-label` date for each day in the calendar will be formed. Uses the same rules/tokens as `dateFormat´ as described here: https://flatpickr.js.org/formatting/. When changing this, make sure the outcome makes sense when using a screenreader.*/
+  ariaDateFormat?: string
+  /** Pass custom classNames. These will be appended to the input element of the DateTimePicker. */
+  className?: string
+  /** A custom string to separate individual dates in `multiple` mode. */
+  conjunction?: string
+  /** A string of characters to customize how a date will be formatted in the input field. Available options: https://flatpickr.js.org/formatting/ */
+  dateFormat?: string
+  /** Sets the default date of the DateTimePicker. Same as `value`, only here for compatibility with the original Flatpickr library. If both `value` and `defaultDate` are being passed, `value` will win. Date Objects, timestamps, ISO date strings, chronological date strings `YYYY-MM-DD HH:MM` (must be compatible to current `dateFormat`), and the shortcut `today` are all accepted. */
+  defaultDate?: DateOption | DateOption[] | null
```

Comment:
**(Blocking Change)**

We don't need to explicitly allow `null` in these props, I think? It's optional, but we didn't support explicit `null` in the past.

Please double check with for all other props too. I see at least five. Removing one causes Storybook file errors, we need to check what was supported in the past etc.

Reply:
I check for nulls and made updates